config/pipeline.php

```diff
@@ -4,13 +4,13 @@
 
 use Psr\Container\ContainerInterface;
 use Zend\Expressive\Application;
+use Zend\Expressive\Handler\NotFoundHandler;
 use Zend\Expressive\Helper\ServerUrlMiddleware;
 use Zend\Expressive\Helper\UrlHelperMiddleware;
 use Zend\Expressive\MiddlewareFactory;
 use Zend\Expressive\Middleware\DispatchMiddleware;
 use Zend\Expressive\Middleware\ImplicitHeadMiddleware;
 use Zend\Expressive\Middleware\ImplicitOptionsMiddleware;
```

Comment:
These are still wrong:

```php
use Zend\Expressive\Router\Middleware\DispatchMiddleware;
use Zend\Expressive\Router\Middleware\ImplicitHeadMiddleware;
use Zend\Expressive\Router\Middleware\ImplicitOptionsMiddleware;
use Zend\Expressive\Router\Middleware\RouteMiddleware;
```

I guess we need another test for this.

Reply:
I don't know that we need a test (I'm not even sure what such a test would look like), but we _do_ need to update any references to classes that were removed or renamed.

Additionally, we need to add the `Zend\Expressive\Router\Middleware\MethodNotFoundMiddleware` into the pipeline, immediately following the entry that pipes the routing middleware.

@samsonasik ^^^

Reply:
I've updated it. it may need an integration test which require the `config/pipeline.php` ?

Reply:
@weierophinney I've added `Zend\Expressive\Router\Middleware\MethodNotAllowedMiddleware` in the `config/pipeline.php` after `RouteMiddleware`

Reply:
We can add a test for the pipeline, but it would pass for the new and old version anyway as the old class names are mapped to the new class names.